test/functional/apps/discover/group3/_time_field_column.ts

```diff
@@ -0,0 +1,455 @@
+                hideTimeFieldColumnSetting,
+                savedSearchSuffix,
+              });
+            });
+
+            it('should render selected columns correctly', async () => {
+              await checkSelectedColumns({
+                hasTimeField: true,
+                hideTimeFieldColumnSetting,
+                savedSearchSuffix,
+              });
+            });
+          });
+
+          describe('without a time field', () => {
+            beforeEach(async () => {
+              await PageObjects.discover.createAdHocDataView('logs*', false);
+              await PageObjects.discover.waitUntilSearchingHasFinished();
+              await PageObjects.header.waitUntilLoadingHasFinished();
+            });
+
+            it('should render initial columns correctly', async () => {
+              await checkInitialColumns({
+                hasTimeField: false,
+                hideTimeFieldColumnSetting,
+                savedSearchSuffix: savedSearchSuffix + '-',
+              });
+            });
+
+            it('should render selected columns correctly', async () => {
+              await checkSelectedColumns({
+                hasTimeField: false,
+                hideTimeFieldColumnSetting,
+                savedSearchSuffix: savedSearchSuffix + '-',
+              });
+            });
+          });
+        });
+
+        describe('ESQL mode', () => {
+          it('should render initial columns correctly', async () => {
+            await PageObjects.discover.selectTextBaseLang();
+
+            await checkInitialColumns({
+              hasTimeField: true,
+              hideTimeFieldColumnSetting,
+              savedSearchSuffix: savedSearchSuffix + 'ESQL',
+              isTextBased: true,
+            });
+          });
+
+          it('should render initial columns correctly when no time field', async () => {
+            await PageObjects.discover.selectTextBaseLang();
+            await monacoEditor.setCodeEditorValue('from logstash-* | limit 10 | drop @timestamp');
+            await testSubjects.click('querySubmitButton');
+            await PageObjects.header.waitUntilLoadingHasFinished();
+
+            await checkInitialColumns({
+              hasTimeField: false,
+              hideTimeFieldColumnSetting,
+              savedSearchSuffix: savedSearchSuffix + 'ESQLdrop',
+              isTextBased: true,
+            });
+          });
+
+          it('should render selected columns correctly', async () => {
+            await PageObjects.discover.selectTextBaseLang();
+
+            await checkSelectedColumns({
+              hasTimeField: true,
+              hideTimeFieldColumnSetting,
+              savedSearchSuffix: savedSearchSuffix + 'ESQL',
+              isTextBased: true,
+            });
+          });
+        });
+
+        // These tests are skipped as they take a lot of time to run. Temporary unskip them to validate current functionality if necessary.
```

Comment:
👍 yes this makes lot's of sense not to test for it, and also it's great that we could if needed